superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -501,22 +529,79 @@ export default class ResultSet extends React.PureComponent<
     return <div />;
   }
 
+  onAlertClose = () => {
+    this.setState({ alertIsOpen: false });
+  };
+
   renderRowsReturned() {
-    const { results, rows, queryLimit } = this.props.query;
-    const limitReached = results?.displayLimitReached;
+    const { results, rows, queryLimit, limitingFactor } = this.props.query;
+    let limitMessage;
+    const isAdmin = this.props.user?.roles.hasOwnProperty('Admin');
+    const adminWarning = isAdmin
+      ? ' by the configuration DISPLAY_MAX_ROWS'
+      : null;
+    if (limitingFactor === LIMITING_FACTOR.QUERY && this.props.csv) {
+      limitMessage = (
+        <span className="limitMessage">
+          {t(
+            `The number of rows displayed is limited to %s by the query`,
+            rows,
+          )}
+        </span>
+      );
+    } else if (
+      limitingFactor === LIMITING_FACTOR.DROPDOWN &&
+      !defaultDropdown(queryLimit, limitingFactor)
+    ) {
+      limitMessage = (
+        <span className="limitMessage">
+          {t(
+            `The number of rows displayed is limited to %s by the limit dropdown.`,
+            rows,
+          )}
+        </span>
+      );
+    } else if (limitingFactor === LIMITING_FACTOR.QUERY_AND_DROPDOWN) {
+      limitMessage = (
+        <span className="limitMessage">
+          {t(
+            `The number of rows displayed is limited to %s by the query and limit dropdown.`,
+            rows,
+          )}
+        </span>
+      );
+    }
     return (
-      <ReturnedRows>
-        {!limitReached && (
-          <Alert type="warning" message={t(`%s rows returned`, rows)} />
-        )}
-        {limitReached && (
+      <ReturnedRows ref={this.alertRef}>
+        {!limitReached(results?.displayLimitReached) &&
+          !defaultDropdown(queryLimit, limitingFactor) && (
+            <span>
+              {t(`%s rows returned`, rows)} {limitMessage}
+            </span>
+          )}
+        {!limitReached(results?.displayLimitReached) &&
+          defaultDropdown(queryLimit, limitingFactor) && (
```

Comment:
do you want to DRY this up and get the value above and assign it to a const to reuse?